include/domain_bridge/domain_bridge_options.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021, Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DOMAIN_BRIDGE__DOMAIN_BRIDGE_OPTIONS_HPP_
+#define DOMAIN_BRIDGE__DOMAIN_BRIDGE_OPTIONS_HPP_
+
+#include <cstddef>
+
+#include "domain_bridge/visibility_control.hpp"
+
+namespace domain_bridge
+{
+
+/// Options for domain bridge.
+class DomainBridgeOptions
+{
+public:
+  /// Constructor.
+  /**
+   *  Default values:
+   *
+   *    - domain_bridge_id = 0u
+   */
+  DOMAIN_BRIDGE_PUBLIC
+  DomainBridgeOptions() = default;
+
+  /// Destructor.
+  DOMAIN_BRIDGE_PUBLIC
+  virtual
+  ~DomainBridgeOptions() = default;
+
+  /// Copy constructor.
+  DOMAIN_BRIDGE_PUBLIC
+  DomainBridgeOptions(const DomainBridgeOptions & other) = default;
+
+  /// Assignment operator.
+  DOMAIN_BRIDGE_PUBLIC
+  DomainBridgeOptions &
+  operator=(const DomainBridgeOptions & other) = default;
+
+  /// Get the identifier for the domain bridge.
+  DOMAIN_BRIDGE_PUBLIC
+  std::size_t
+  id() const;
+
+  /// Set the identifier for the domain bridge.
+  /**
+   * The identifier is added to names of nodes created by the domain bridge.
+   */
+  DOMAIN_BRIDGE_PUBLIC
+  DomainBridgeOptions &
+  id(std::size_t id);
+
+private:
+  std::size_t id_{0u};
```

Comment:
This should IMO be generated by the implementation and not specifed by the user.

Reply:
This was added to avoid duplicate node name in cases where the user creates two or more domain bridges. I think giving the user a way to identifier the bridge nodes is better so they can deterministically predict the names. If we randomly generate node names, then I imagine this would make it more difficult to enable ROS security.

I think this is kind of equivalent to giving a node a name. Maybe renaming the option to "name" makes more sense, what do you think?

Reply:
And the generated node names could be "{options.name()}_{domain_id}".

Reply:
That convinces me, an user provided name sounds fine.

Reply:
Refactored to use a string, `name`, instead of an ID (2379dab)